grails-app/services/org/pih/warehouse/picklist/PicklistService.groovy

```diff
@@ -74,12 +75,13 @@ class PicklistService {
                 it.disableRefresh = Boolean.TRUE
                 picklist.removeFromPicklistItems(it)
                 orderItem.removeFromPicklistItems(it)
+                binLocations.add(it.binLocation?.id)
```

Comment:
Since we're not doing any filtering (like in cases above) we could probably just collect all of the bin locations before the findAll.
```
def binLocationIds = picklist?.picklistItems?.collect { it?.binLocation?.id }
```